lib/debug/breakpoint.rb

```diff
@@ -253,6 +253,8 @@ def setup
       @tp = TracePoint.new(:line){|tp|
         next if tp.path.start_with? __dir__
         next if tp.path.start_with? '<internal:'
+        # Skip when `require 'json'` and `JSON.generate` are called during tests
+        next if tp.path.start_with? RbConfig.expand '$(libdir)'
```

Comment:
doesn't this mean all Ruby stdlib will be skipped? this changes how debugger works so probably should have more explanation.
it's also not clear to me that why `JSON.generate` would trigger the `CheckBreakpoint`. we may want to fix this instead of skipping all the stdlib.

Reply:
> doesn't this mean all Ruby stdlib will be skipped? this changes how debugger works so probably should have more explanation.

Exactly. I changed it.

> it's also not clear to me that why JSON.generate would trigger the CheckBreakpoint.

FYI, `if State === opts` in `generate` method is one of the triggers in this tracepoint.
https://github.com/flori/json/blob/master/lib/json/common.rb#L297

Reply:
good catch. could you give me a call-stack when it is called? Maybe it should be change the test framework design.

Reply:
```shell
["/Users/[user_name]/workspace/debug/lib/debug/breakpoint.rb:14:in `safe_eval'", "/Users/[user_name]/workspace/debug/lib/debug/breakpoint.rb:260:in `block in setup'", "/Users/[user_name]/.rbenv/versions/3.0.0/lib/ruby/3.0.0/json/common.rb:297:in `generate'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:193:in `wait_command'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:177:in `block (2 levels) in wait_command_loop'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:176:in `loop'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:176:in `block in wait_command_loop'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:816:in `block in stop_all_threads'", "<internal:trace_point>:196:in `enable'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:815:in `stop_all_threads'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:175:in `wait_command_loop'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:116:in `session_server_main'", "/Users/[user_name]/workspace/debug/lib/debug/session.rb:75:in `block in initialize'"]
```

Reply:
Are they what you want?

Reply:
I'll fix it later.